examples/kmeans.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include "lib/ml/data_loader.hpp"
+#include "lib/ml/kmeans.hpp"
+#include "lib/vector.hpp"
+
+#include "core/engine.hpp"
+
+using LabeledPointHObj = husky::lib::ml::LabeledPointHObj<double, int, false>;
+
+void testKmeans() {
+
+    int maxIter = std::stoi(husky::Context::get_param("iter"));
+
+    // 1. read data
+    auto& point_list = husky::ObjListStore::create_objlist<LabeledPointHObj>("point_list");
+    int dim = husky::lib::ml::load_data(husky::Context::get_param("input"), point_list, husky::lib::ml::kLIBSVMFormat);
+    husky::LOG_I << "local point_list size " << point_list.get_data().size();
+
+    // 2. train
+    auto kmeansOp = husky::lib::ml::Kmeans<double, int, false>(3,maxIter); //template arg: feature is a double non-sparse vec, label is int (not used)
```

Comment:
3 is no. of centers in Kmeans, maxIter is the number of iteration to do.
They can be omitted in the constructor and defaults to 1.

The template args <double, int, false> matches the data from load_data,
meaning a LabeledPoint with a int label, and a non-sparse double vector.
I agree these template args are not ideal, but right now I don't know enough 
C++ template tricks to avoid them